packages/core/src/internal/proc.js

```diff
@@ -462,7 +462,7 @@ export default function proc(env, iterator, parentContext, parentEffectId, meta,
     asap(() => {
       let result
       try {
-        result = (channel ? channel.put : env.dispatch)(action)
+        result = channel.put(action)
```

Comment:
`env.dispatch` is removed here. So running sagas do all the input/output via a channel.